lcftrans/src/translation.cpp

```diff
@@ -119,8 +118,49 @@ Translation Translation::Merge(const Translation& from) {
 	return stale;
 }
 
-template<typename T, typename U, typename F>
-static void parseEvents(Translation& t, lcf::StringView parent_name, U& root, const F& make_info) {
+using map_event_ctx = const lcf::Context<lcf::rpg::EventCommand,
+		lcf::Context<lcf::rpg::EventPage,
+		lcf::Context<lcf::rpg::Event,
+		lcf::Context<lcf::rpg::Map,
+		lcf::Context<void, void>>>>>;
+
+using common_event_ctx = const lcf::Context<lcf::rpg::EventCommand,
+		lcf::Context<lcf::rpg::CommonEvent,
+		lcf::Context<lcf::rpg::Database,
+		lcf::Context<void, void>>>>;
+
+using battle_event_ctx = lcf::Context<lcf::rpg::EventCommand,
+		lcf::Context<lcf::rpg::TroopPage,
+		lcf::Context<lcf::rpg::Troop,
+		lcf::Context<lcf::rpg::Database,
+		lcf::Context<void, void>>>>>;
+
+template <typename T> bool isEventCommandString(const lcf::ContextStructBase<T>&) {	return false; }
+bool isEventCommandString(const lcf::ContextStructBase<lcf::rpg::EventCommand>& ctx) { return ctx.name == "string"; }
+
+template<typename T>
+int getEventId(const T&) { return -1; }
+int getEventId(const map_event_ctx& ctx) { return ctx.parent->parent->obj->ID; }
+int getEventId(const common_event_ctx& ctx) { return ctx.parent->obj->ID; }
+int getEventId(const battle_event_ctx& ctx) { return ctx.parent->parent->obj->ID; }
+
+template<typename T>
+lcf::Span<lcf::rpg::EventCommand> getEventCommands(const T&) { return {}; }
+lcf::Span<lcf::rpg::EventCommand> getEventCommands(const map_event_ctx& ctx) { return ctx.parent->obj->event_commands; }
+lcf::Span<lcf::rpg::EventCommand> getEventCommands(const common_event_ctx& ctx) { return ctx.parent->obj->event_commands; }
+lcf::Span<lcf::rpg::EventCommand> getEventCommands(const battle_event_ctx& ctx) { return ctx.parent->obj->event_commands; }
```

Comment:
Spezialisation foo for ``getEventId`` and ``getEventCommands`` because this depends on the event :)

Reply:
There are other ways to make this more generic, more templately, and less readable. Give there are only 3 cases what you have here is perfectly fine.